### PR TITLE
roger - no ticket - Fileuploadv2

### PR DESCRIPTION
**Pull request by roggc**
multiple file upload in parallel, v2

https://drive.google.com/file/d/1lCx5mXvAC0TkYSCC3kVJ65FbwsuMB-cQ/view?usp=sharing

dropArea
Screenshot 2022-01-13 at 08 48 54
Screenshot 2022-01-13 at 08 49 23

### CODE REVIEW

**Review by ralph-dev — COMMENTED** [review]
It's on the right track but the dropArea has a lot of logic that should be exposed as props

**Comment by roggc** [comment]
@ralph-dev , what logic? I don't see any logic ...

File: src/Containers/DropArea/DropArea.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { Meta, Story } from '@storybook/react';
+import { DropArea, IDropAreaProps } from './DropArea';
+
+export default {
+    title: 'Components/DropArea',
+    component: DropArea,
```
Comment (ralph-dev) [inline src/Containers/DropArea/DropArea.stories.tsx:7]: argTypes: {onClick: action('File Panel Was Clicked')}

File: src/Containers/FileUploadV2/FileUploadV2.tsx [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+                (panel) => panel.name === name,
+            );
+            if (informativePanel) {
+                if (
+                    base64StringFile === NO_BASE64STRINGFILE ||
+                    isTestIsFailure
+                ) {
+                    setInformativePanels((prev) => ({
+                        ...prev,
+                        panels: prev.panels.map((panel) => {
```
Comment (ralph-dev) [inline src/Containers/FileUploadV2/FileUploadV2.tsx:121]: abstract into function getPanels(OperationStateParam)

File: src/Containers/FileUploadV2/FileUploadV2.tsx [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+                            if (panel.name === informativePanel.name)
+                                return {
+                                    ...panel,
+                                    operationState: OperationState.isSuccess,
+                                };
+                            return panel;
+                        }),
+                        makeItDisappear: [
```
Comment (ralph-dev) [inline src/Containers/FileUploadV2/FileUploadV2.tsx:146]: duplicate

File: src/Containers/FileUploadV2/FileUploadV2.tsx [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+            }
+            const informativePanel = informativePanels.panels.find(
+                (panel) => panel.name === name,
+            );
+            if (informativePanel) {
+                if (
+                    base64StringFile === NO_BASE64STRINGFILE ||
+                    isTestIsFailure
+                ) {
+                    setInformativePanels((prev) => ({
```
Comment (ralph-dev) [inline src/Containers/FileUploadV2/FileUploadV2.tsx:119]: Abstract this first, remove duplicate

File: src/Containers/Panel/Panel.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                panel.worker?.terminate();
+            }
+        },
+        [],
+    );
+
+    return (
+        <Card {...props}>
+            <div>{getMessage()}</div>
+            {panel?.operationState === OperationState.isLoading && (
```
Comment (ralph-dev) [inline src/Containers/Panel/Panel.tsx:40]: replace with isLoading prop

Comment (ralph-dev) [reply src/Containers/Panel/Panel.tsx:40]: isLoading = True
is Loading = False

isSuccess = null
isError = null

isSuccess= 'text' | isSuccess=''completed''

File: src/Containers/Panel/Panel.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import React, { useCallback, useEffect } from 'react';
+import styled from 'styled-components';
+import { IPanelProps, OperationState } from '../FileUploadV2/FileUploadV2';
+import { Card as C } from '../Card/Card';
+import { Button as B } from '../../Inputs/Button/Button';
+
```
Comment (ralph-dev) [inline src/Containers/Panel/Panel.tsx:6]: interface

File: src/Containers/Panel/Panel.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+            case OperationState.isLoading:
+                return `loading ${panel.name} ...`;
+            default:
+                return '';
+            }
+        }
+        return '';
+    }, [panel]);
+
+    useEffect(
```
Comment (ralph-dev) [inline src/Containers/Panel/Panel.tsx:28]: remove

File: src/Containers/Panel/Panel.tsx (hunk 6 above)
Comment (ralph-dev) [reply src/Containers/Panel/Panel.tsx:6]: interface IPanel {
 /* Cancel Upload */
 onCancel
}

**Comment by ralph-dev** [comment]
Let's close this as well @roggc

**Pull request closed without merging**